src/app/hooks/WithTermsAndConditions.tsx

```diff
@@ -0,0 +1,85 @@
+import { ComponentType, useState, useCallback } from "react";
+import { useTranslation } from "react-i18next";
+import { TermsAndConditionModal } from "@rhoas/app-services-ui-components";
+import { useModalControl } from "@app/hooks";
+import { ITermsConfig } from "@app/services";
+import { useConstants } from "@app/providers/config/ServiceConstants";
+
+export enum ServiceType {
+  KAFKA,
+  SERVICE_REGISTRY,
+  CONNECTORS,
+}
+
+export function withTermsAndConditions<P>(
+  WrappedComponent: ComponentType<P>,
+  serviceType: ServiceType
+) {
+  const getServiceName = () => {
+    switch (serviceType) {
+      case ServiceType.KAFKA:
+        return "common:Kafka";
+      case ServiceType.SERVICE_REGISTRY:
+        return "srsTemporaryFixMe:srs.service_registry";
+      default:
+        return "common:Kafka";
+    }
+  };
+
+  return (props: P): JSX.Element => {
```

Comment:
I'm not convinced of this. I think it will work, but the TS signature isn't correct for a React component. A React component isn't just a function returning JSX, there is a bit more behind that.

If the returned function didn't have to use hooks, this would probably be ok. But given that it has too, I'm not sure.

Why a HOC? Can't this be a simple hook that components need to use? Something like:

```ts
const TermsAndConditionsBoundary = useTermsAndConditionsBoundary(t('common:kafka'));

return (
  <TermsAndConditionsBoundary>
    ...
  </TermsAndConditionsBoundary>
)
```

Reply:
There is a issue by using hook, how can we get the callback functions (handlePreCreateInstance, shouldOpenCreateModal) in parent component KasPage from hook which return jsx. 
any suggestion ?